src/install.c

```diff
@@ -608,35 +611,37 @@ static void prepare_environment(GSubprocessLauncher *launcher, gchar *update_sou
 		}
 
 		varname = g_strdup_printf("RAUC_SLOT_NAME_%i", slotcnt);
-		g_subprocess_launcher_setenv(launcher, varname, slot->name, TRUE);
+		envp = g_environ_setenv(envp, varname, slot->name, TRUE);
 		g_clear_pointer(&varname, g_free);
 
 		varname = g_strdup_printf("RAUC_SLOT_CLASS_%i", slotcnt);
-		g_subprocess_launcher_setenv(launcher, varname, slot->sclass, TRUE);
+		envp = g_environ_setenv(envp, varname, slot->sclass, TRUE);
 		g_clear_pointer(&varname, g_free);
 
 		varname = g_strdup_printf("RAUC_SLOT_TYPE_%i", slotcnt);
-		g_subprocess_launcher_setenv(launcher, varname, slot->type, TRUE);
+		envp = g_environ_setenv(envp, varname, slot->type, TRUE);
 		g_clear_pointer(&varname, g_free);
 
 		varname = g_strdup_printf("RAUC_SLOT_DEVICE_%i", slotcnt);
-		g_subprocess_launcher_setenv(launcher, varname, slot->device, TRUE);
+		envp = g_environ_setenv(envp, varname, slot->device, TRUE);
 		g_clear_pointer(&varname, g_free);
 
 		varname = g_strdup_printf("RAUC_SLOT_BOOTNAME_%i", slotcnt);
-		g_subprocess_launcher_setenv(launcher, varname, slot->bootname ? slot->bootname : "", TRUE);
+		envp = g_environ_setenv(envp, varname, slot->bootname ? slot->bootname : "", TRUE);
 		g_clear_pointer(&varname, g_free);
 
 		varname = g_strdup_printf("RAUC_SLOT_PARENT_%i", slotcnt);
-		g_subprocess_launcher_setenv(launcher, varname, slot->parent ? slot->parent->name : "", TRUE);
+		envp = g_environ_setenv(envp, varname, slot->parent ? slot->parent->name : "", TRUE);
 		g_clear_pointer(&varname, g_free);
 	}
 
-	g_subprocess_launcher_setenv(launcher, "RAUC_SLOTS", slots->str, TRUE);
-	g_subprocess_launcher_setenv(launcher, "RAUC_TARGET_SLOTS", target_slots->str, TRUE);
+	envp = g_environ_setenv(envp, "RAUC_SLOTS", slots->str, TRUE);
+	envp = g_environ_setenv(envp, "RAUC_TARGET_SLOTS", target_slots->str, TRUE);
+
+	return envp;
 }
 
-static gboolean launch_and_wait_handler(RaucInstallArgs *args, gchar *update_source, gchar *handler_name, RaucManifest *manifest, gchar **handler_argv, GHashTable *target_group, GError **error)
+static gboolean launch_and_wait_handler(RaucInstallArgs *args, gchar *handler_name, gchar **handler_argv, gchar **setenv, GError **error)
```

Comment:
`setenv` is a function from `stdlib.h`, so we should not use it as a parameter name.